installer/manifests/keptn/core.yaml

```diff
@@ -1,32 +1,230 @@
+            # auth via backend (if the subrequest returns a 2xx response code, the access is allowed. If it returns 401 or 403,
+            # the access is denied) before we store the file
+            # see http://nginx.org/en/docs/http/ngx_http_auth_request_module.html
+            auth_request               /v1/auth;
+
+            rewrite /configuration-service/(.*) /$1  break;
+            proxy_pass         http://configuration-service:8080;
+            proxy_redirect     off;
+            proxy_set_header   Host $host;
+            proxy_http_version 1.1;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+          }
+
+          # special configuration for /v1/auth to always use POST requests
+          location /v1/auth {
+            rewrite /v1/auth/(.*) /$1  break;
+
+            proxy_method POST;
+            proxy_pass_request_body off;
+
+            proxy_pass http://api-service:8080;
+            proxy_http_version 1.1;
+            proxy_pass_request_headers on; # only pass reuquest headers to this service
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+          }
+
+          location /v2 {
+            # ToDo
+            rewrite /v2/(.*) /$1  break;
+          }
+
+          # default location: use existing api
+          location / {
+            proxy_pass http://api-service:8080;
+            proxy_http_version 1.1;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection "Upgrade";
+            proxy_set_header Host $host;
+          }
+      }
+    }
 ---
-# api
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: api
+  name: api-gateway-nginx
   namespace: keptn
 spec:
   selector:
     matchLabels:
-      run: api
+      run: api-gateway-nginx
   replicas: 1
   template:
     metadata:
       labels:
-        run: api
+        run: api-gateway-nginx
     spec:
       containers:
-        - name: api
-          image: keptn/api:latest
+        - name: api-gateway-nginx
+          image: nginx:1.17.9
           ports:
-            - containerPort: 8080
+            - containerPort: 80
+          volumeMounts:
+            - mountPath: /etc/nginx # mount api-nginx-config volume to /etc/nginx
+              readOnly: true
+              name: api-nginx-config
           resources:
             requests:
               memory: "64Mi"
               cpu: "50m"
             limits:
               memory: "128Mi"
               cpu: "500m"
```

Comment:
we should double check the actual requirements for API gateway in terms of cpu and memory.

Reply:
We reduced the limits.